zrythm: 1.0.0-alpha.28.1.3 -> 1.0.0-beta.4.5.62; carla: 2.5.1 -> 2.5.3

yuuyins opened the pull request:
###### Description of changes

zrythm: 1.0.0-alpha.28.1.3 
carla: 2.5.1 -> 2.5.3
zix: init at unstable-2023-02-13

supersedes
- https://github.com/NixOS/nixpkgs/pull/190730
- https://github.com/NixOS/nixpkgs/pull/175505
- https://github.com/NixOS/nixpkgs/pull/196227

tested and works

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

yuuyins committed:
zix: init at unstable-2023-02-13

Co-authored-by: zseri <zseri.devel@ytrizja.de>

yuuyins committed:
maintainers: change email, add matrix

yuuyins committed:
zrythm: 1.0.0-alpha.28.1.3 -> 1.0.0-beta.4.5.62

Co-authored-by: zseri <zseri.devel@ytrizja.de>
Co-authored-by: Alexandros Theodotou <alex@zrythm.org>
Co-authored-by: Sandro Jäckel <sandro.jaeckel@gmail.com>

Zrythm 1.0.0-beta.4.5.62 needs
https://github.com/falkTX/Carla/tree/main/source/includes/clap, which is only
available on unstable.

yuuyins committed:
carla: 2.5.1 -> 2.5.3

Co-authored-by: Alexandros Theodotou <alex@zrythm.org>
Co-authored-by: Bart Brouns <bart@magnetophon.nl>
Co-authored-by: Sandro Jäckel <sandro.jaeckel@gmail.com>

cole-h commented:
@ofborg eval

magnetophon commented on pkgs/applications/audio/carla/default.nix:
```diff
@@ -13,13 +13,13 @@ assert withGtk3 -> gtk3 != null;
 stdenv.mkDerivation rec {
   pname = "carla";
-  version = "2.5.1";
+  version = "unstable-2023-02-12";
```
Did https://github.com/falkTX/Carla/releases/tag/v2.5.3 not work?

magnetophon commented:
Maybe it would be best to make Carla unstable a separate pkg.

magnetophon reviewed (approved):
Looks great, thanks!

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/847

SuperSandro2000 commented on pkgs/development/libraries/audio/zix/default.nix:
```diff
@@ -0,0 +1,44 @@
+, pkg-config
+}:
+
+let
+  pname = "zix";
+  version = "unstable-2023-02-13";
+
+  src = fetchFromGitLab {
+    owner = "drobilla";
+    repo = pname;
+    rev = "262d4a1522c38be0588746e874159da5c7bb457d";
+    hash = "sha256-3vuefgnirM4ksK3j9sjBHgOmx0JpL+6tCPb69/7jI00=";
+  };
+in stdenv.mkDerivation {
+  inherit pname version src;
```
```suggestion
stdenv.mkDerivation rec {
  pname = "zix";
  version = "unstable-2023-02-13";

  src = fetchFromGitLab {
    owner = "drobilla";
    repo = pname;
    rev = "262d4a1522c38be0588746e874159da5c7bb457d";
    hash = "sha256-3vuefgnirM4ksK3j9sjBHgOmx0JpL+6tCPb69/7jI00=";
  };
```
we can make this a bit more straight forward

Lord-Valen replied:
If you don't want to use `rec` you could also do
```nix
let
  name = "zix";
in stdenv.mkDerivation rec {
  pname = name;
  version = "unstable-2023-02-13";
  src = fetchFromGitLab {
    owner = "drobilla";
    repo = name;
    rev = "262d4a1522c38be0588746e874159da5c7bb457d";
    hash = "sha256-3vuefgnirM4ksK3j9sjBHgOmx0JpL+6tCPb69/7jI00=";
  };
```

Lord-Valen replied:
> i suppose it was not your intention to have left rec in in stdenv.mkDerivation rec { in your suggestion?

Oh I didn't notice that I had left `rec` in.	Oops!

> i can't see how the suggestion would improve over inherit (i mean the pname=name bits)

In my opinion, minimizing the contents of a `let` block is typically more readable. It's more immediately apparent what is being given to `mkDerivation`.

> i'm using src.rev on meta.changelog as well for this derivation, and i could always have to use version as well...

If you want to also use rev, you could hoist that up to the `let` block as well.

> i know this rec / no rec may be a minor thing, but I really want to avoid anything that might cause those nix issues.

You're using the `let` block like a `rec` here *anyway* (you're using it for self-reference). The end result was that you added extra boilerplate/clutter (the `inherit`s) to no added benefit. That's why Sandro recommended those changes. Arguably, this use of a `let` block is even more of an anti-pattern than `rec` since it's really just `rec` with extra steps. If you look closely, what nix.dev was suggesting as an alternative to `rec` was exactly the sort of thing I suggested: extract the commonalities to new variables, and put those variables in scope with a `let` block.

All that being said, there isn't really a risk of inifinite recursion using `rec` this way. Either go for the `rec` as Sandro suggested, or use my suggested `let` block. Both are acceptable imo.

SuperSandro2000 commented on pkgs/applications/audio/zrythm/default.nix:
```diff
@@ -161,13 +192,17 @@ stdenv.mkDerivation rec {
-    description = "Highly automated and intuitive digital audio workstation";
-    maintainers = with maintainers; [ tshaynik magnetophon ];
+    description = "Automated and intuitive digital audio workstation";
+    maintainers = with maintainers; [
+      tshaynik
+      magnetophon
+    ];
```
```suggestion
    maintainers = with maintainers; [ tshaynik magnetophon ];
```
nit
we don't need to change this. Doing it like this is fine.

SuperSandro2000 commented on pkgs/applications/audio/zrythm/default.nix:
```diff
@@ -161,13 +192,17 @@ stdenv.mkDerivation rec {
     gappsWrapperArgs+=(
       --prefix GSETTINGS_SCHEMA_DIR : "$out/share/gsettings-schemas/${pname}-${version}/glib-2.0/schemas/"
       --prefix XDG_DATA_DIRS : "$XDG_ICON_DIRS"
```
```suggestion
      --prefix XDG_DATA_DIRS : "$XDG_ICON_DIRS:${breeze-icons}/share"
```
we can combine this to generate less code

SuperSandro2000 commented on pkgs/applications/audio/zrythm/default.nix:
```diff
@@ -122,36 +139,50 @@ stdenv.mkDerivation rec {
     rtaudio
     rtmidi
     rubberband
+    sassc
```
Isn't sassc a css compiler that is run on the host?

zseri replied:
no I think this is about why it is in `buildInputs` and not in `nativeBuildInputs`

SuperSandro2000 commented on pkgs/applications/audio/carla/default.nix:
```diff
@@ -1,91 +1,13 @@
-    license = licenses.gpl2Plus;
-    maintainers = [ maintainers.minijackson ];
-    platforms = platforms.linux;
+  carla-unstable = libsForQt5.callPackage ./common.nix {
+    version = "unstable-2023-02-12";
+    hash = "sha256-OegwW8sfUV4IjBFrq38tpKbOMzZT0cvhL/u3Gr1m938=";
+    rev = "4c29367b6e6f82e75746b9ee3a01f9d205c0beae";
```
Do we only need this for one package? Then we should not expose it on top-level and rather just do the overwrite for the one package

SuperSandro2000 commented on pkgs/applications/audio/carla/common.nix:
```diff
@@ -0,0 +1,109 @@
+  buildInputs = [
+    file liblo alsa-lib fluidsynth jack2 libpulseaudio libsndfile
+  ] ++ lib.optional withQt qtbase
+    ++ lib.optional withGtk2 gtk2
```
We should think about removing gtk2

minijackson replied:
IIRC, gtk2 is needed for plugins whose interface is implemented using gtk2. I can't think of an example plugin right now, but removing it might break the setup of some people using gtk2-based audio plugins

SuperSandro2000 commented on pkgs/applications/audio/carla/common.nix:
```diff
@@ -0,0 +1,109 @@
+assert withFrontend -> python3Packages ? pyqt5;
+assert withQt -> qtbase != null;
+assert withQt -> wrapQtAppsHook != null;
+assert withGtk2 -> gtk2 != null;
+assert withGtk3 -> gtk3 != null;
```
```suggestion
```
that will never be the case in nixpkgs

SuperSandro2000 commented on pkgs/applications/audio/carla/common.nix:
```diff
@@ -0,0 +1,109 @@
+, hash
+}:
+
+assert withFrontend -> python3Packages ? pyqt5;
```
```suggestion
```
same here

Lord-Valen reviewed (approved):
Looks good!

zseri commented on pkgs/applications/audio/carla/default.nix:
```diff
@@ -1,25 +1,38 @@
 
   src = fetchFromGitHub {
-    owner = "falkTX";
+    owner = "falktx";
```
why was this changed?

zseri commented on pkgs/applications/audio/zrythm/default.nix:
```diff
@@ -46,31 +55,48 @@
+    version = "unstable-2023-02-24";
+
+    src = fetchFromGitHub {
+      owner = "falktx";
```
```suggestion
      owner = "falkTX";
```

minijackson reviewed (approved):
Works great for carla, thanks!

SuperSandro2000 commented on pkgs/applications/audio/carla/default.nix:
```diff
@@ -1,25 +1,38 @@
+, pkg-config
+, python3Packages
+, which
+, gtk2 ? null
+, gtk3 ? null
```
```suggestion
```

SuperSandro2000 reviewed (changes requested):
Please move the gitignore changes to another PR unless they are directly related to the updated packages but then we might as well put a gitignore file into that specific directory.
Changes to that file are relevant for a lot of people and I cannot just merge them in a package bump.

SuperSandro2000 committed:
carla: remove python3Packages and gtk2, gk3 assertions

According to Nixpkgs maintainer Sandro Jäckel,

> that will never be the case in nixpkgs
> ---Sandro Jäckel, https://github.com/NixOS/nixpkgs/pull/217424#discussion_r1117657269
> ---Sandro Jäckel, https://github.com/NixOS/nixpkgs/pull/217424#discussion_r1117657635

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/866
